collection-renderer-lib/src/cmr/collection_renderer/services/collection_renderer.clj

```diff
@@ -122,10 +123,31 @@
        (search/find-concepts context :granule)
        :hits))
 
+(defn- get-services-for-collection
+  "Retrieve service association for the collection corresponding to concept-id."
+  [context concept-id]
+  (let [collection-results (->> {:concept-id concept-id
+                                 :result-format {:format :umm-json-results}}
+                                (params/parse-parameter-query context :collection)
+                                (search/find-concepts context :collection)
+                                :results)
+        items (:items (json/parse-string collection-results true))
+        service-ids (get-in (first items) [:meta :associations :services])]
+    (for [service-id service-ids
+          :let [service-results (->> {:concept-id service-id
+                                      :result-format {:format :umm-json-results}}
+                                     (params/parse-parameter-query context :service)
+                                     (search/find-concepts context :service)
+                                     :results)
+                items (:items (json/parse-string service-results true))]]
+      (first items))))
```

Comment:
Nitpick: maybe add a comment about why you're only taking the first item

Reply:
One service-id will return at most one record.